synthesizer/program/src/resources/credits.aleo

```diff
@@ -106,573 +149,515 @@ record credits:
 
 /**********************************************************************************************************************/
 
-// This function allows any staker to bond their microcredits to a validator and specify a withdrawal address.
-// The corresponding functions for 'bond_public' are 'unbond_public' and 'claim_unbond_public'.
-function bond_public:
-    // Input the validator's address.
-    input r0 as address.public;
-    // Input the withdrawal address.
-    input r1 as address.public;
-    // Input the amount of microcredits to bond.
-    input r2 as u64.public;
-
-    // Determine if the amount is at least one credit.
-    gte r2 1_000_000u64 into r3;
-    // Enforce the amount is at least one credit.
+// This function allows a validator to self-bond to join the active committee, and specify a commission percent and withdrawal address.
+// Commission percent is the portion of delegator rewards that the validator claims for itself.
+// A commission of 100 would result in all delegator rewards being retained by the validator.
+// bond_validator can also be used to increase the self-bond of the validator.
+function bond_validator:
+    input r0 as address.public;  // Withdrawal address.
+    input r1 as u64.public;      // Amount of microcredits to bond.
+    input r2 as u8.public;       // Commission percent.
+
+    // Prevent using hot-on-box validator address from withdrawing self-bond and rewards.
+    assert.neq self.signer r0;
+
```

Comment:
We need to prevent the validator from being a program address, like we did in - https://github.com/AleoNet/snarkVM/pull/2443.

Fortunately, I believe since this function is validator specific, a simple `assert.neq self.signer self.caller` would suffice.